### ISSUE TITLE
Datetimepicker UI component is not accessible to screen-reader and keyboard users

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#5325]
### Issue Summary

The current datetime picker UI is very difficult to use for people relying on keyboards for navigation, screen reader users but also sighted users with motor skill impairments. Here is the current widget for reference:

<img width="539" alt="datetime-picker" src="https://user-images.githubusercontent.com/877585/58487056-e1665600-815e-11e9-9dd4-db11146d9375.png">

(under the hood this is https://github.com/xdan/datetimepicker)

Here are specific issues with this implementation:

- When tabbing to the datetime fields and the picker opens, it doesn't announce for screen reader users that it has been opened.
- There is no way to move keyboard focus within the datetime picker UI to navigate through it with the keyboard
- Even if it was possible to navigate with the keyboard in this UI, most/all of the buttons are icon-only and don't have a screen-reader-friendly label
- Colour contrast fail – should darken pink highlighted colour 10%: #DA665E

### Steps to Reproduce

1. Go to the "Settings" tab of any Wagtail page
2. Try to interact with the "Go-live" or "Expiry" fields

---

I can see a few solutions to this, not sure which it makes sense to aim for:

- Leave it as it is, and do what we can to make it easier for people to enter date-times as text directly in the text field (for example, adding validation, and some help text with the expected format)
- Rework the current widget to better manage focus and keyboard navigation, and to have SR-only labels
- Switch to another widget, for example https://github.com/airbnb/react-dates
- Split the datetime fields into two widgets – one for dates, one for times.

**Comment by thibaudcolas** [comment]
From @katielocke3 on this:

> I’m in favour of switching to a newer widget UI. This would not only improve accessibility but also usability for all users.
> 
> The airbnb example is great however it doesn’t show the time of the publishing date, not sure if this is something that should say within the UI?
> 
> Another good example is Google Calendars date picker UI which shows the date as well as time which you can type within the field or select from a drop down list. 

![time-picker](https://user-images.githubusercontent.com/877585/59360483-10b4bf80-8d28-11e9-8789-33f6cebf84c3.gif)

**Comment by thibaudcolas** [comment]
Suggestion: https://tempusdominus.github.io/bootstrap-4/

**Comment by thibaudcolas** [comment]
https://axesslab.com/accessible-datepickers/

**Comment by FlipperPA** [comment]
** Breaking for lighting talks, will return to edit this comment soon! **

More information on Tempus Dominus. Tempus Dominus is the successor and total-rewrite of the popular bootstrap3-datepicker, and has separated the core date/time picker code from the rendering layer. You can see the code for core, bootstrap3, and bootstrap 4 here:

It contains many configurable options:

https://tempusdominus.github.io/bootstrap-4/Options/

Keybinds might be useful:

https://tempusdominus.github.io/bootstrap-4/Options/#keybinds

There are some open accessibility issues under bootstrap 4:

https://github.com/tempusdominus/bootstrap-4/issues/257

**Comment by thibaudcolas** [comment]
More relevant bits of info:

- Good round up of accessible options, h/t to @nicklee for sharing https://www.digitala11y.com/accessible-datepickers-roundup/
- Safari TP 115 released last week has support for the HTML `<input type="date">`, so we may well be able to use this a few years from now. https://webkit.org/blog/11333/release-notes-for-safari-technology-preview-115/ h/t @balazs-endresz for the heads’ up!
